elastic/kibana
# [Upgrade Assistant]  Add deep-link to Cloud's upgrade modal from the Upgrade step

Fixes https://github.com/elastic/kibana/issues/114050

### Summary

This PR adds a flag to the `upgrade on cloud` link that allows the user to navigate to the deployment page and have the upgrade modal opened up by default (whereas before it was just to the deployments page and the user had to find and click the link)

**How to test**
1. Start up elasticsearch, `yarn es snapshot --license=trial`
2. Add the following to `config/kibana.yml`:
```yml
xpack.cloud.id: 'eastus2.azure.elastic-cloud.com:9243$59ef636c6917463db140321484d63cfa$a8b109c08adc43279ef48f29af1a3911'
xpack.cloud.base_url: "https://cloud.elastic.co"
xpack.cloud.deployment_url: "/deployments/bfdad4ef99a24212a06d387593686d63"
```
3. Start up Kibana, `yarn start` and navigate to UA
4. Verify that the url from the `Upgrade on Cloud` link has the `?show_upgrade=true` query param.
